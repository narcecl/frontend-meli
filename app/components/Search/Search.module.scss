.MeliSearch{
    width: 100%;
    position: relative;

    &__input{
        padding: 6px 48px 6px 12px;
        width: 100%;
        border: 0;
        font-size: 1.125em;
        color: #333;
        font-family: "Proxima Nova", -apple-system, Roboto, Arial, sans-serif, sans-serif;
        border-radius: 4px;
        box-sizing: border-box;

        &:focus, &:focus-within{
            outline: 1px solid #3483fa;
        }
    }

    &__button{
        position: absolute;
        right: 0;
        top: 0;
        width: 38px;
        height: 100%;
        border: none;
        cursor: pointer;
        border-radius: 0 4px 4px 0;
    }
}