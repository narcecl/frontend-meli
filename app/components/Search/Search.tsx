import React from 'react';
import styles from './Search.module.scss';

export interface SearchProps {
    value: string;
    placeholder?: string;
    onInput: (value: string) => void;
}

export const Search = (props: SearchProps) => {
    const { value, placeholder, onInput } = props;

    const handleInput = (e: any): void => {
        onInput(e.target.value);
    }

    return (
        <form action="/items" method="GET" className={styles.MeliSearch}>
            <input
                className={styles.MeliSearch__input}
                onInput={handleInput}
                value={value}
                name="search"
                placeholder={placeholder}
            />
            <button
                className={styles.MeliSearch__button}
                aria-label="Buscar"
                type="submit"
            />
        </form>
    );
}

export default Search;