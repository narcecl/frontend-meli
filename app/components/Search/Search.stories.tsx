import type { Meta, StoryObj } from '@storybook/react';
import { Search } from './Search';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta: Meta<typeof Search> = {
    title: 'Core/Search',
    component: Search,
    tags: ['autodocs'],
    argTypes: {
        value: {
            description: 'Defines input state (Getter).',
        },
        placeholder: {
            description: 'Defines input native placeholder attribute',
        },
        onInput: {
            description: 'Defines onInput event on input',
            control: false,
        }
    }
};

export default meta;
type Story = StoryObj<typeof Search>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
    parameters: {
        backgrounds: { default: 'primary' },
    },
    args: {
    },
};