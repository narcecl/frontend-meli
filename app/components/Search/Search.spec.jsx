import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Search from './Search';

describe('Search', () => {
    it('Default render', () => {
        // Mock event
        const onInput = jest.fn();

        // Render the component
        const { container, asFragment } = render(
            <Search
                value=""
                onInput={onInput}
                placeholder="Nunca dejes de buscar"
            />
        );

        // Check first render
        const firstRender = asFragment();
        expect(firstRender).toMatchSnapshot();

        // Check input and attributes
        const input = container.querySelector('input');
        expect(input).toBeInTheDocument();
        expect(input).toHaveAttribute('placeholder', 'Nunca dejes de buscar');
    });

    it('Test onInput', () => {
        // Mock event
        const mockFn = jest.fn();

        // Render the component
        const { container, asFragment } = render(
            <Search
                value=""
                onInput={mockFn}
                placeholder="Nunca dejes de buscar"
            />
        );

        // Check first render
        const firstRender = asFragment();
        expect(firstRender).toMatchSnapshot();
        
        // Check input and attributes
        const input = container.querySelector('input');
        expect(input).toBeInTheDocument();
        expect(input).toHaveAttribute('placeholder', 'Nunca dejes de buscar');
        
        // Check mocked onInput event
        fireEvent.input(input, { target: { value: 'new value' } });
        expect(mockFn).toHaveBeenLastCalledWith('new value');
    })
});