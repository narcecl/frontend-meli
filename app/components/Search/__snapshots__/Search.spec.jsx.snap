// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Default render 1`] = `
<DocumentFragment>
  <form
    action="/items"
    class="MeliSearch"
    method="GET"
  >
    <input
      class="MeliSearch__input"
      name="search"
      placeholder="Nunca dejes de buscar"
      value=""
    />
    <button
      aria-label="Buscar"
      class="MeliSearch__button"
      type="submit"
    />
  </form>
</DocumentFragment>
`;

exports[`Search Test onInput 1`] = `
<DocumentFragment>
  <form
    action="/items"
    class="MeliSearch"
    method="GET"
  >
    <input
      class="MeliSearch__input"
      name="search"
      placeholder="Nunca dejes de buscar"
      value=""
    />
    <button
      aria-label="Buscar"
      class="MeliSearch__button"
      type="submit"
    />
  </form>
</DocumentFragment>
`;
